### PR TITLE
Fixed #25858 -- Bound abstract model application relative relationships.

### PR DESCRIPTION
**Pull request by charettes**
Thanks to Karl Hobley for the report and Markus, Shai, Aymeric for their input.

### CODE REVIEW

File: tests/model_fields/tests.py [hunk 1]
```diff
@@ -251,6 +251,25 @@ class Meta:
             "Pending lookup added for a foreign key on an abstract model"
         )
 
+    @isolate_apps('model_fields', 'model_fields.tests')
```
Comment (charettes) [inline tests/model_fields/tests.py]: Will make sure to use an explicit `Apps()` instance when backporting.

File: tests/model_fields/tests.py [hunk 2]
```diff
@@ -273,6 +292,30 @@ class Meta:
             "Pending lookup added for a many-to-many field on an abstract model"
         )
 
+    @isolate_apps('model_fields', 'model_fields.tests')
```
Comment (charettes) [inline tests/model_fields/tests.py]: Ditto.

File: django/db/models/fields/related.py [hunk 3]
```diff
@@ -301,6 +301,13 @@ def resolve_related_class(model, related, field):
                 field.remote_field.model = related
                 field.do_related_class(related, model)
             lazy_related_operation(resolve_related_class, cls, self.remote_field.model, field=self)
+        else:
+            # Make sure lazy reference is bound to an application.
```
Comment (timgraham) [inline django/db/models/fields/related.py]: Is it correct to say, "Bind a lazy reference to the app in which the model is defined."
Maybe a helper method is worth avoiding some repetition?

File: docs/releases/1.9.2.txt [hunk 4]
```diff
@@ -30,3 +30,7 @@ Bugfixes
   ``db_index=True`` or ``unique=True`` to a ``CharField`` or ``TextField`` that
   already had the other specified, or when removing one of them from a field
   that had both (:ticket:`26034`).
+
+* Fixed a regression with abstract models defining application relative lazy
```
Comment (timgraham) [inline docs/releases/1.9.2.txt]: This is a tough one to describe, but here's my try: "Fixed a regression where defining a relation on an abstract model's field using a string model name without an app_label no longer resolved that reference to the abstract model's app if using that model in another app"

File: django/db/models/fields/related.py (hunk 3 above)
Comment (charettes) [inline django/db/models/fields/related.py]: Yeah I hard trouble coining the correct terms, will change.

**Pull request merged**